spec/models/user_spec.rb

```diff
@@ -1,5 +1,36 @@
 require 'rails_helper'
 
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+describe User do
+  it { should have_valid(:provider).when("github") }
+  it { should_not have_valid(:provider).when(nil, "", "facebook") }
+  it { should have_valid(:uid).when("1234") }
+  it { should_not have_valid(:uid).when(nil, "") }
+
+  describe ".from_omniauth" do
+    it "should return a newly created user if they are not in the database" do
+      expect { User.from_omniauth(mock_github_auth!) }.to change{ User.count }.from(0).to(1)
+      expect(User.from_omniauth(mock_github_auth!)).to be_a(User)
+    end
+
+    it "should return an existing user if they are in the database" do
+      user = User.from_omniauth(mock_github_auth!)
+      expect { User.from_omniauth(mock_github_auth!) }.not_to change{ User.count }
+      expect(User.from_omniauth(mock_github_auth!)).to eq(user)
+    end
+  end
+
+  describe "#update_from_omniauth" do
+    let(:user) { create(
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.